hedera-node/hedera-app/src/test/java/com/hedera/node/app/blocks/impl/BlockStreamManagerImplTest.java

```diff
@@ -494,6 +513,14 @@ private void givenSubjectWith(
             @NonNull final PlatformState platformState,
             @NonNull final BlockItemWriter... writers) {
         given(round.getConsensusTimestamp()).willReturn(CONSENSUS_NOW);
+        given(round.iterator()).willReturn(eventIterator);
+        given(eventIterator.hasNext()).willReturn(true, false);
+        given(eventIterator.next()).willReturn(event);
+        given(event.consensusTransactionIterator()).willReturn(transactionIterator);
+        given(transactionIterator.hasNext()).willReturn(true).willReturn(true).willReturn(false);
+        given(transactionIterator.next()).willReturn(transaction1).willReturn(transaction2);
+        given(transaction1.getConsensusTimestamp()).willReturn(null);
+        given(transaction2.getConsensusTimestamp()).willReturn(CONSENSUS_NOW);
```

Comment:
Not necessary for this PR, but I think we could make this easier to read by creating an actual collection of transactions and an actual collection of events, then using `.iterator()` instead of mocking the iterators themselves. Just a thought

Reply:
Yes that is an option. Since it was just checking if the consensus time was used, didn't find reasonable to construct the real objects.